build-scripts/collect_remediations.py

```diff
@@ -0,0 +1,117 @@
+        help="YAML file with information about the build configuration. "
+        "e.g.: ~/scap-security-guide/build/build_config.yml"
+    )
+    p.add_argument(
+        "--product-yaml", required=True,
+        help="YAML file with information about the product we are building. "
+        "e.g.: ~/scap-security-guide/rhel7/product.yml"
+    )
+    p.add_argument(
+        "--resolved-rules-dir", required=True,
+        help="Directory with <rule-id>.yml resolved rule YAMLs"
+    )
+    p.add_argument(
+        "--remediation-type", required=True, action="append",
+        help="language or type of the remediations we are combining."
+        "example: ansible")
+    p.add_argument(
+        "--output-dir", required=True,
+        help="output directory where all remediations will be saved"
+    )
+    p.add_argument(
+        "--fixes-from-templates-dir", required=True,
+        help="directory from which we will collect fixes generated from "
+        "templates")
+
+    return p.parse_args()
+
+
+def prepare_output_dirs(output_dir, remediation_types):
+    output_dirs = dict()
+    for lang in remediation_types:
+        language_output_dir = os.path.join(output_dir, lang)
+        if not os.path.exists(language_output_dir):
+            os.makedirs(language_output_dir)
+        output_dirs[lang] = language_output_dir
+    return output_dirs
+
+
+def combine_remediations(
+        rule, langs, fixes_from_templates_dirs, product, output_dirs,
+        env_yaml):
+    for lang in langs:
+        rule_dir = os.path.dirname(rule.definition_location)
+        ext = remediation.REMEDIATION_TO_EXT_MAP[lang]
+        remediation_cls = remediation.REMEDIATION_TO_CLASS[lang]
+        language_fixes_from_templates_dir = os.path.join(
+            fixes_from_templates_dirs, lang)
+        fix_path = None
+        # first look for a static remediation
+        rule_dir_remediations = remediation.get_rule_dir_remediations(
+            rule_dir, lang, product)
+        if len(rule_dir_remediations) > 0:
+            # first item in the list has the highest priority
+            fix_path = rule_dir_remediations[0]
+        if fix_path is None:
+            # check if we have a templated remediation instead
+            if os.path.isdir(language_fixes_from_templates_dir):
+                templated_fix_path = os.path.join(
+                    language_fixes_from_templates_dir, rule.id_ + ext)
+                if os.path.exists(templated_fix_path):
+                    fix_path = templated_fix_path
+        if fix_path is None:
+            # neither static nor templated remediation found
+            continue
+        remediation_obj = remediation_cls(fix_path)
+        remediation_obj.associate_rule(rule)
+        fix = remediation.process(remediation_obj, env_yaml)
+        if fix:
+            output_file_path = os.path.join(output_dirs[lang], rule.id_ + ext)
+            remediation.write_fix_to_file(fix, output_file_path)
```

Comment:
I propose to extract this to a function as well - given a path, we either write the fix into a directory, or we don't do anything.